src/actions/actions.ts

```diff
@@ -2246,6 +2249,48 @@ class MoveToMatchingBracket extends BaseMovement {
   }
 }
 
+abstract class MoveInsideCharacter extends MoveToMatchingBase {
+  modes = [ModeName.Normal, ModeName.Visual, ModeName.VisualLine];
+  protected charToMatch: string;
+
+  public async execAction(position: Position, vimState: VimState): Promise<Position | IMovement> {
+    const text = TextEditor.getLineAt(position).text;
+    const pairedCharacter = this.pairings[this.charToMatch]
+    const charToClose = pairedCharacter ? pairedCharacter.match : this.charToMatch;
+
+    // First, search backwards for the opening character of the sequence
+    let start = -1;
+    for (let i = position.character; i >= 0; i--) {
+      if (text[i] == this.charToMatch) {
+        start = i;
+        break;
+      }
+    }
+    const startPos = new Position(position.line, start + 1);
+    const endPos = this.nextBracket(startPos, this.charToMatch, { match: charToClose, nextMatchIsForward: true }, false);
```

Comment:
This is where the behavior differs from vim. This should be calling some variant of `nextBracket` that only iterates positions up to the end of the current line, not the whole document.